src/parser/clauses/index.h

```diff
@@ -0,0 +1,21 @@
+#ifndef _CLAUSE_INDEX_H
+#define _CLAUSE_INDEX_H
+
+typedef enum {
+  DROP_INDEX,
+  CREATE_INDEX
+} AST_IndexOpType;
+
+typedef struct {
+  const char *label;
+  const char *property;
+} IndexTarget;
+
+typedef struct {
+  IndexTarget target;
+  AST_IndexOpType operation;
+} AST_IndexNode;
+
+AST_IndexNode* AST_IndexOp(const char *label, const char *property, AST_IndexOpType optype);
```

Comment:
Please use a function name similar to the other clauses, see create clause for example.